lib/phoenix/pubsub/pg2.ex

```diff
@@ -0,0 +1,58 @@
+defmodule Phoenix.PubSub.PG2 do
+  use GenServer
+
+  @moduledoc """
+  PubSub adapter based on `:pg2`
+  """
+
+  @private_pg2_group {:phx, :global}
+
+
+  def start_link(opts) do
+    GenServer.start_link __MODULE__, [], name: Dict.fetch!(opts, :name)
+  end
+
+  def init(_opts) do
+    {:ok, local_pid} = Phoenix.PubSub.Local.start_link()
+    :ok = :pg2.create(@private_pg2_group)
+    :ok = :pg2.join(@private_pg2_group, self)
+    {:ok, %{local_pid: local_pid}}
+  end
+
+  def handle_call({:subscribe, pid, topic}, _from, state) do
+    {:reply, GenServer.call(state.local_pid, {:subscribe, pid, topic}), state}
```

Comment:
Btw, instead of `{:redirect, pid}`, we should do `{:client, module, fun, args}`. This is because we mostly want to execute this in the client instead of blocking the server. Moving to the MFA approach will allow us to be more generic, allowing @pma to also optimize his code even though he is not using local. It would also allow us to optimize the broadcast and connection pool usage in the Redis one.